File: scripts/convert_pdf_to_image.py
import os
import subprocess

def convert_pdfs_to_images():
    # Define directories
    pdf_dir = os.path.join('public', 'pdfs', 'conveyor')
    output_dir = os.path.join('public', 'images', 'projects', 'conveyor')
    
    # Create output directory if it doesn't exist
    os.makedirs(output_dir, exist_ok=True)
    
    # Convert each PDF
    for pdf_file in os.listdir(pdf_dir):
        if pdf_file.endswith('.pdf'):
            pdf_path = os.path.join(pdf_dir, pdf_file)
            output_base = os.path.join(output_dir, pdf_file[:-4])  # Remove .pdf extension
            
            try:
                # Convert first page of PDF to PNG using ImageMagick
                subprocess.run([
                    'magick',
                    'convert',
                    '-density', '300',
                    '-quality', '100',
                    f'{pdf_path}[0]',
                    f'{output_base}.png'
                ])
                print(f"Converted {pdf_file} to PNG")
            except Exception as e:
                print(f"Error converting {pdf_file}: {str(e)}")

if __name__ == "__main__":
    convert_pdfs_to_images()

File: scripts/convertPdfToImage.js
const fs = require('fs');
const path = require('path');
const pdf2img = require('pdf-img-convert');

async function convertPDFsToImages() {
    const pdfDir = path.join(__dirname, '..', 'public', 'pdfs', 'conveyor');
    const outputDir = path.join(__dirname, '..', 'public', 'images', 'projects', 'conveyor');

    // Create output directory if it doesn't exist
    if (!fs.existsSync(outputDir)) {
        fs.mkdirSync(outputDir, { recursive: true });
    }

    // Get all PDF files
    const pdfFiles = fs.readdirSync(pdfDir).filter(file => file.endsWith('.pdf'));

    for (const pdfFile of pdfFiles) {
        const pdfPath = path.join(pdfDir, pdfFile);
        const outputBase = path.join(outputDir, pdfFile.replace('.pdf', ''));

        try {
            const pdfArray = await pdf2img.convert(pdfPath, {
                width: 1920,
                height: 1080,
                page_numbers: [1]
            });

            // Save the first page as PNG
            fs.writeFileSync(`${outputBase}.png`, pdfArray[0]);
            console.log(`Converted ${pdfFile} to PNG`);
        } catch (error) {
            console.error(`Error converting ${pdfFile}:`, error);
        }
    }
}

convertPDFsToImages().catch(console.error);
